FrankenDrift.GlkRunner/FrankenDrift.GlkRunner/GlkHtmlWin.cs

```diff
@@ -107,50 +107,48 @@ public void Clear()
             GlkApi.glk_window_clear(glkwin_handle);
         }
 
-        internal unsafe string GetLineInput()
+        internal string GetLineInput()
         {
             if (IsWaiting)
                 throw new ConcurrentEventException("Too many input events requested");
             IsWaiting = true;
             const uint capacity = 256;
-            var cmdToBe = new uint[capacity];
-            fixed (uint* buf = cmdToBe)
```

Comment:
I'm not convinced that this is a safe thing to do. (Nevermind that you can't pass `Memory` to P/Invoke, see other commend.) Using `Memory` does not implicitly pin the underlying array in memory.

The two P/Invoke-related examples [here](https://learn.microsoft.com/en-us/dotnet/standard/memory-and-spans/memory-t-usage-guidelines) (9 and 10) both use either `fixed` or `Memory.Pin` and pass the resulting raw pointers to P/Invoke.

Reply:
I had seen that page. I didn't think #10 applied to our situation as GetLineInput is essentially synchronous, and the usual C# GC should determine the correct time to clean up the Memory variable. But maybe `fixed` is required.

Reply:
The problem with GC isn't that the buffer will be *deleted* (we keep our own reference, after all), but that it might decide to *move* the array in the time between calling glk\_request\_line\_event and the event actually occurring. (In theory, any number of timer, sound, or rearrange events could be processed in the meantime.)